csrc/scheduler/vectorize_helper.h

```diff
@@ -72,91 +73,74 @@ class TORCH_CUDA_CU_API ProjectedExtent {
   // provided value.
   void multiplyNumeratorValue(Val* new_numerator_val) {
     TORCH_INTERNAL_ASSERT(
-        !new_numerator_val->isZeroInt() &&
-            (!new_numerator_val->isConstInt() ||
-             new_numerator_val->evaluateInt() > 0),
-        "Adding numerator value of zero not supported in ProjectedExtent.");
-
-    zero_ = false;
-    if (new_numerator_val->isConstInt()) {
-      const_numerator_vals_.push_back(new_numerator_val->evaluateInt());
+        quotient_ == nullptr && is_divisible_ == nullptr,
+        "Can not modify finalized ProjectedExtent.");
+    if (numerator_ == nullptr) {
+      numerator_ = new_numerator_val;
     } else {
-      symb_numerator_vals_.push_back(new_numerator_val);
+      numerator_ = SimplifyingIrBuilder::mulExpr(numerator_, new_numerator_val);
     }
-    valid_extents_ = false;
-    is_simplified_ = false;
+
+    zero_ = false;
```

Comment:
Do we still need this?

Reply:
Also, the comment about this class seems to need to be updated. 

> but once a value is added it can never be zero again.

Reply:
I have no idea, I am just preserving the previous behavior in this PR. I will look into zero tensor support soon after this cleanup. Created an issue for now: https://github.com/NVIDIA/Fuser/issues/410

Reply:
Looks like this was the only use of `zero_`:

https://github.com/NVIDIA/Fuser/pull/400/files#diff-52fd86634438c842c38ba0ad2c46247ac85b309cdcf345981494492e0b97fd1fL264

It seems that was just used to distinguish an initial state. 

Previously, when `zero_` is true, the numerator was zero. Now, that seems to be different and it's one:

https://github.com/NVIDIA/Fuser/blob/main/csrc/scheduler/vectorize_helper.h#L120-L123

Not sure if this difference has any impact.

Reply:
I think we also have code like:
```C++
void ContiguousInnerDimensionsMapper::propagateExtentMergeBackward(
    const Merge* merge) {
  auto out_mapping = getMappedExtent(merge->out());
  if (out_mapping.isZero()) {
    return;
  }
```

Reply:
OK, missed that.

Shouldn't the quotient be zero when `zero_` is true?

https://github.com/NVIDIA/Fuser/blob/main/csrc/scheduler/vectorize_helper.h#L134

Reply:
I think now `numerator_ == nullptr` and `zero_` is always the same, and we are treating the numerator as `1`. So unless the denominator is also `1`, the quotient will be zero.

Reply:
So, the quotient of an empty `ProjectedExtent` is now `1`, although its `zero_` is true.

Also, even when the denominator is not `1`, unless all the multiplication components of the denominator are constant, the denominator will be symbolic, so the quotient would be something like `1 / symbolic_val`. Here, `zero_` is still true, but `quotient()->isZeroInt()` is false.

As long as `ProjectedExtent::isZero()` is always used to filter out "zero" extents, this inconsistency doesn't seem to cause any issue.

Reply:
> So, the quotient of an empty `ProjectedExtent` is now `1`, although its `zero_` is true.

Yes, correct.

> Also, even when the denominator is not `1`, unless all the multiplication components of the denominator are constant, the denominator will be symbolic, so the quotient would be something like `1 / symbolic_val`. Here, `zero_` is still true, but `quotient()->isZeroInt()` is false.

You are right. `quotient()->isZeroInt()` is false, but after evaluation, `quotient()` might evaluate to zero.

> As long as `ProjectedExtent::isZero()` is always used to filter out "zero" extents, this inconsistency doesn't seem to cause any issue.

Correct, I don't expect any behavior change, because `quotient()` is never used when the PE is "zero", although it is not clear to me what "zero" means...